pkg/machines/hostvmslist.es6

```diff
@@ -0,0 +1,299 @@
+  );
+}
+
+function StateIcon ({ state } ) {
+  function iconElement ({onClick, className, title}) {
+      return React.createElement("span", {title, 'data-toggle': 'tooltip', 'data-placement': 'left'}, `${state} `,
+        React.createElement('i', {onClick, className}));
+  }
+
+  switch (state) {
+    case 'running':// TODO: display VM screenshot if available or the ok-icon otherwise
+      return iconElement({className: 'pficon pficon-ok icon-1x-vms', title: 'The VM is running.'});
+    case 'idle':
+      return iconElement({className: 'pficon pficon-running icon-1x-vms', title: 'The VM is idle.'});
+    case 'paused':
+      return iconElement({className: 'pficon pficon-pause icon-1x-vms', title: 'The VM is paused.'});
+    case 'shutdown':
+      return iconElement({className: 'glyphicon glyphicon-wrench icon-1x-vms', title: 'The VM is going down.'});
+    case 'shut off':
+      return iconElement({className: 'fa fa-arrow-circle-o-down icon-1x-vms', title: 'The VM is down.'});
+    case 'crashed':
+      return iconElement({className: 'pficon pficon-error-circle-o icon-1x-vms', title: 'The VM crashed.'});
+    case 'dying':
+      return iconElement({className: 'pficon pficon-warning-triangle-o icon-1x-vms', title: 'The VM is in process of dying (shut down or crash is not completed).'});
+    case 'pmsuspended':
+      return iconElement({className: 'pficon pficon-ok icon-1x-vms', title: 'The VM is suspended by guest power management'});
+    case undefined:
+      return React.createElement("div");
+    default:
+      return React.createElement("small", null, state);
+  }
+}
+
+/**
+ * Render group of buttons as a dropdown
+ *
+ * @param buttons array of objects [ {title, action, id}, ... ].
+ *        At least one button is required. Button id is optional.
+ * @returns {*}
+ * @constructor
+ */
+function DropdownButtons(buttons) {
+  const buttonsHtml = buttons.map(
+    button => React.createElement("li", {className: "presentation"},
+      React.createElement("a", {role: "menuitem", onClick: button.action, id: button['id']}, button.title))
+  );
+
+  const caretId = buttons[0]['id'] ? `${buttons[0]['id']}-caret` : undefined;
+
+  return React.createElement("div", {className: "btn-group"},
+    React.createElement("button", {className: "btn btn-default btn-danger", onClick: buttons[0].action}, buttons[0].title),
+    React.createElement("button", {'data-toggle': "dropdown", className: "btn btn-default dropdown-toggle"},
+      React.createElement("span", {className: "caret", id: caretId})),
+    React.createElement("ul", {role: "menu", className: "dropdown-menu"}, buttonsHtml)
+  );
+}
+
+function vmId ( vmName ) {
+  return `vm-${vmName}`;
+}
+const VmOverviewTab = React.createClass({
+  propTypes: {
+    vm: React.PropTypes.object.isRequired,
+  },
+  render: function() {
+    const vm = this.props.vm;
+
+    return React.createElement("table", {className: "machines-width-max"},
```

Comment:
using jsx for these files might make them easier to read, but that can be done in a follow-up